### PR TITLE
[4.0] Improve Menu/Toolbar RTL in atum

### PR DESCRIPTION
**Pull request by wilsonge**
Improves the menu (text is now on the right not left of the menu), and the toolbar (things now render semi-correctly). The table options drop down also now opens correctly below the dropdown

@ciar4n please review me trying to implement CSS - I'm sure its dreadful in more than one way :)

Test by installing the Persian Language Pack

### CODE REVIEW

**Comment by infograf768** [comment]
MUCH better!

We also need in template-rtl
`.main-nav .fa{margin:0 18px} ` // same as LTR

to get
screen shot 2018-07-16 at 08 41 18

instead of 
`.main-nav .fa{margin:0 12px 0 8px}`

screen shot 2018-07-16 at 08 40 54

File: administrator/templates/atum/scss/template-rtl.scss [hunk 1]
```diff
@@ -226,3 +229,17 @@ th {
...
+
+.js-stools .js-stools-container-filters {
+  left: 0;
+  right: auto;
+}
+
+.ml-auto, .mx-auto {
+  margin-left: 0 !important;
+  margin-right: auto !important;
+}
```
[the remark this answers is not in the document]
Comment (ciar4n) [reply administrator/templates/atum/scss/template-rtl.scss]: Try an avoid changing Bootstrap utility classes in the template-rtl.css. Flex is RTL by default so should be no need for this.

[the remark this answers is not in the document]
Comment (ciar4n) [reply administrator/templates/atum/scss/template-rtl.scss]: Changing the following to `flex` will make these automatically RTL.. https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/blocks/_sidebar.scss#L84

**Comment by ciar4n** [comment]
Delete.. https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/template-rtl.scss#L45

Not needed and pushing menu out form edge of viewport

File: administrator/templates/atum/css/template-rtl.css [hunk 2]
```diff
@@ -1714,6 +1709,12 @@ th {
   left: 15px;
   content: "\f0d9"; }
 
+.main-nav li a span {
+  float: right; }
```
Comment (ciar4n) [inline administrator/templates/atum/css/template-rtl.css:1714]: Once you change https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/blocks/_sidebar.scss#L84 to flex, remove this. Floats are ignored in flex objects.

**Comment by ciar4n** [comment]
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/template-rtl.scss#L118

Change to `right: 60px` for correct position of sub menu

**Comment by ciar4n** [comment]
As mentioned by @infograf768, remove the following... https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/template-rtl.scss#L111-L113

**Comment by wilsonge** [comment]
> Change to right: 60px for correct position of sub menu

Ahh that's the one I was looking for but couldn't find :) Thanks!

I've also simplified down the toolbar a bit more. So with these corrections. Any objections to merging this? It's far from perfect but much better than before :P

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 2f186aeb36fb0ab2d6ab473342ed16d7f39c80d2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21136">issues.joomla.org/tracker/joomla-cms/21136</a>.</sub>

**Comment by wilsonge** [comment]
One last bug fix for the top menu 😇

**Comment by infograf768** [comment]
If you can, let's also correct this

screen shot 2018-07-16 at 13 24 48

```
.mr-2, .mx-2 {
    margin-right: .5rem !important;
}
```
Normally if I understand well scss
it should be in LTR
```
.mr-2,
.mx-2 {
  margin-right: 0.5rem !important; }
 margin-left: 0;  
}
```
and in RTL
```
.mr-2,
.mx-2 {
    margin-right: 0;
    margin-left: .5rem !important;
}
```

**Comment by wilsonge** [comment]
@infograf768 Which view is this in?

**Comment by wilsonge** [comment]
Don't worry - found it and done

**Comment by infograf768** [comment]
Articles manager

File: administrator/templates/atum/scss/template-rtl.scss [hunk 3]
```diff
@@ -160,38 +160,29 @@ th {
 // Toolbar
 .subhead {
 
-  .btn-group-sm > .btn {
+  .btn, .btn-sm {
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/template-rtl.scss:163]: You can remove the `.btn-sm` from here as it's an addition to `.btn` which is required anyway

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 04d3851144dbad2b10fa9ad48a839b4ff997e8f9<br /><br />Good to go for now. Solves a lot of stuff. If we find something missing, we can add issue later.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21136">issues.joomla.org/tracker/joomla-cms/21136</a>.</sub>

**Comment by laoneo** [comment]
@franz-wohlkoenig can you also retest please.

**Comment by ghost** [comment]
@laoneo will do in next Minutes.

**Comment by ghost** [comment]
@infograf768 this for another Issue?

screen shot 2018-07-16 at 17 26 27
screen shot 2018-07-16 at 17 26 09

**Comment by wilsonge** [comment]
Yah there's a huge amount more to do here. But let's take it one step at a time :)

**Comment by infograf768** [comment]
Note: for the cpanel
.flex-grow-1 is missing a text-align: right;
.list-group is also missing text-align: right;

**Comment by C-Lodder** [comment]
Don't add it to `flex-grow-1`. This class instructs the element to fill available space and will do so regardless of RTL

**Comment by infograf768** [comment]
Indeed
list-group is enough

**Comment by wilsonge** [comment]
With the merge of https://github.com/joomla/joomla-cms/pull/20962 that's not required for now as they've been moved to tables

**Pull request merged**